src/mysky/index.ts

```diff
@@ -313,6 +315,57 @@ export class MySky {
     return { data: json, dataLink };
   }
 
+  /**
+   * Deletes Discoverable JSON at the given path through MySky, if the user has given Write permissions to do so.
+   *
+   * @param path - The data path.
+   * @param [customOptions] - Additional settings that can optionally be set.
+   * @returns - An object containing the json data as well as the skylink for the data.
+   */
+  async deleteJSON(path: string, customOptions?: CustomSetJSONOptions): Promise<void> {
+    validateString("path", path, "parameter");
+    validateOptionalObject("customOptions", customOptions, "parameter", defaultSetJSONOptions);
+
+    const opts = {
+      ...defaultSetJSONOptions,
+      ...this.connector.client.customOptions,
+      ...customOptions,
+    };
+
+    const publicKey = await this.userID();
+    const dataKey = deriveDiscoverableTweak(path);
+    opts.hashedDataKeyHex = true; // Do not hash the tweak anymore.
+
+    // Fetch the current value to find out the revision.
+    const getEntryOpts = extractOptions(opts, defaultGetEntryOptions);
+    const signedEntry = await this.connector.client.registry.getEntry(publicKey, dataKey, getEntryOpts);
+
+    // TODO: Refactor
+    let revision: bigint;
+    if (signedEntry.entry === null) {
+      revision = BigInt(0);
+    } else {
+      revision = signedEntry.entry.revision + BigInt(1);
+    }
+
+    // Throw if the revision is already the maximum value.
+    if (revision > MAX_REVISION) {
+      throw new Error("Current entry already has maximum allowed revision, could not update the entry");
+    }
+
+    // Build the registry value.
+    const entry: RegistryEntry = {
+      dataKey,
+      data: emptySkylink,
```

Comment:
I read this as setting the data to the string "AAA..." and not a skylink v2 uint8[] of [0,0,0,0,...] which is what that Base64 string should be. Is this right?

Reply:
Yeah I'll have to set it to the bytes in order for skyd to recognize it, and for it to work with v2 skylinks. The v2 skylink fix PR is a prerequisite for that.